### ISSUE TITLE
port 8000 camunda cockpit auto timeout in 30 sec after login

### ISSUE THREAD

**Issue by man-chi** [AOT-Technologies/forms-flow-ai#155]
we have setup the AWS environment with all stacks running such as formsform frontend, formsform backend, dashboard and camunda service. all the rest of services are running fine, however after successfully login using "formsform-designer" user to the port 8000 (camunda cockpit). within 30sec, the authentication to cockpit page will be timed out and redirected to the keycloak login page again. 

steps to reproduce:
* access: http://<internal_ip>:8000/camunda/app/cockpit/
* login with "formsform-designer" user
* browse the process for a few seconds
* auth session timeout and redirected to login page   <-- problem

please help and advise

**Comment by zmezei-proxifile** [comment]
The error message we see in the Keycloak docker container's log is this:

```
08:31:18,329 WARN  [org.keycloak.events] (default task-317) type=REFRESH_TOKEN_ERROR, realmId=forms-flow-ai, clientId=forms-flow-bpm, userId=null, ipAddress=172.22.0.1, error=invalid_token, grant_type=refresh_token, client_auth_method=client-secret
```

It seems that `userId` is set to null, which might not be OK.

**Comment by man-chi** [comment]
hi aot members, 

due to some reason, userId is set to null , can you kindly advise on this?

**Comment by shibin-aot** [comment]
In keycloak set the redirect uri as per the below screenshot under forms-flow-bpm clien
![camunda-keycloak](https://user-images.githubusercontent.com/83489861/135797476-cf296a19-9641-444f-b628-83c5c1fb1ea7.PNG)
t.

**Comment by man-chi** [comment]
@shibin-aot we have already setup the redirect URI before (as shown in the picture below), can you kindly help to take a look on what could have been missing still
<img width="1174" alt="Screen Shot 2021-10-04 at 3 40 34 PM" src="https://user-images.githubusercontent.com/84918340/135812235-778621f0-780a-491a-b06f-f06dfeb3f590.png">
?

**Comment by shibin-aot** [comment]
Add localhost / host name also along with the redirect uri

**Comment by man-chi** [comment]
@shibin-aot is the setting like this?
<img width="767" alt="Screen Shot 2021-10-04 at 5 17 12 PM" src="https://user-images.githubusercontent.com/84918340/135825702-ee89e9c9-f146-48f2-a4f2-bc2bce11c129.png">

**Comment by man-chi** [comment]
after access the camunda cockpit for 10sec, it still always timeout, redirecting to login page. http://10.2.50.133:8000/camunda/app/cockpit/default/#/tasks 
<img width="1413" alt="Screen Shot 2021-10-04 at 5 18 55 PM" src="https://user-images.githubusercontent.com/84918340/135826377-5c698d02-29ca-44a4-b1dc-787915f1cae3.png">

**Comment by shibin-aot** [comment]
Yes the localhost is correct, you can remove the other 2 and access camunda using localhost / hostname instead of ip

**Comment by man-chi** [comment]
I have tried to set redirect URI to be "http://localhost:8000/camunda/*"
however,  encountered error immediately when trying to access camunda cockpit url below.
http://10.2.50.133:8000/camunda/app/cockpit/

<img width="991" alt="Screen Shot 2021-10-04 at 6 00 26 PM" src="https://user-images.githubusercontent.com/84918340/135832351-114bdee2-4aa3-413c-83f1-7c4f80628539.png">

**Comment by zmezei-proxifile** [comment]
From the context of the running Camunda instance, Keycloak on "localhost" is probably not available. Keycloak and Camunda are started from different docker-compose files, so they create different networks. For this to work, I think the host IP is required for hte communication.

**Comment by man-chi** [comment]
@zmezei-proxifile , should we assign a hostname/domain name to private IP  10.2.50.133 and try to use internal domain instead?

**Comment by man-chi** [comment]
@shibin-aot please help to advise. localhost in redirect URI does not work , having an error. please kindly advise on the next step

**Comment by shibin-aot** [comment]
Please refer to this screenshots, it should work.
![camunda](https://user-images.githubusercontent.com/83489861/135835549-7cbfbe71-ede4-44bd-84b4-5dfd4c94e258.PNG)
![cockpit](https://user-images.githubusercontent.com/83489861/135835559-48552917-6057-49c4-8bbf-8f9ea550f528.PNG)

**Comment by man-chi** [comment]
@shibin-aot I tried the localhost and * like you have mentioned above but unfortunately, it still timeout after 10 sec, redirecting to login page.

@zmezei-proxifile possible to get keycloak and camunda running on the same network under  localhost or any private domain name?

**Comment by shibin-aot** [comment]
Let me know if you are available to join for a quick google connect, so we can resolve the issue. shibin.thomas@aot-technologies.com is my email id. Let me know your availability, I can schedule one meeting.

**Comment by man-chi** [comment]
I am in Asian timezone +8GMT; Zoltan, Devops expert, is in Central European +2 GMT ,

as a result, our best overlap timeslote will be from 7am PST to 9am PST daily.  how about you?


Manchi Leung
VP Engineering | Proxifile Inc.
Claims as a Service 



+886-921935451 <tel:510-910-3838> | Line ID: manchi.leung

***@***.*** ***@***.***>
	www.proxifile.com <http://www.proxifile.com/> 

9F No. 287, Sec. 3, Nanjing E. Rd., Songshan Dist., Taipei City <https://g.page/SkyCo-NF?share>

**Comment by shibin-aot** [comment]
We work on IST, so 10 AM IST to 7 PM IST I will be available for calls. You can mail me with your availability

**Comment by EderGuima** [comment]
Hey guys!
Almost same problem here. But we arnt been redirect with timeout instead when we click some where on page.
Our console log are returning 400, 401, 403, 404

![image](https://user-images.githubusercontent.com/48599995/135920726-ce7fd11c-6516-4c38-9360-450819e8279a.png)

and then...

![image](https://user-images.githubusercontent.com/48599995/135920811-640a1f90-12f0-46ff-a173-2a039cd85f00.png)

To login again only restarting all services.

Some Warning are shown as the file below
[log camunda.txt](https://github.com/AOT-Technologies/forms-flow-ai/files/7281166/log.camunda.txt)

We are using docker.
